app/pages/learn/01_tutorial/04_mastering-the-api/02_invoke/00_methodhandle.md

```diff
@@ -0,0 +1,626 @@
+decide whether we should limit its access level.
+
+For example, if we create a method handle pointing to a private method and that method handle is accessible from outside,
+so is the private method. Normally we would like to avoid that. One way is to make the lookup object and method handle
+`private` too. Another option is to create the lookup object using [`MethodHandles.publicLookup`](javadoc:MethodHandles.publicLookup())
+method, so it will only be able to search for public members in public classes within packages that are exported unconditionally:
+
+```java
+MethodHandles.Lookup publicLookup = MethodHandles.publicLookup();
+```
+
+If we are going to keep the lookup object and the method handles private, it's safe to give them access to any members,
+including private and protected ones:
+
+```java
+MethodHandles.Lookup lookup = MethodHandles.lookup();
+```
+
+<a id="methodtype">&nbsp;</a>
+## Method type
+To look up a method handle we also need to provide the type information of the method or field. The method
+type information is represented as [`MethodType`](javadoc:MethodType) object. To instantiate a `MethodType`,
+we have to provide the return type as the first parameter followed by all the argument types:
+
+```java
+MethodType methodType = MethodType.methodType(int.class /* the method returns integer */,
+        String.class /* and accepts a single String argument*/);
+```
+
+Having the `Lookup` and the `MethodType` instances, we can look up the method handle. For instance methods, we should
+use [`Lookup.findVirtual`](javadoc:MethodHandles.Lookup.findVirtual(Class,String,MethodType)), and for static methods
+[`Lookup.findStatic`](javadoc:MethodHandles.Lookup.findStatic(Class,String,MethodType)). Both these methods accept the
+following arguments: a `Class` where the method is located, a method name represented as a `String`, and a `MethodType`
+instance.
+
+In the example below, we are using `Lookup.findVirtual` method to look up an instance method
+[`String.replace`](javadoc:String.replace(char,char)), which accepts two `char` arguments and returns a `String`:
+
+```java
+MethodHandles.Lookup lookup = MethodHandles.lookup();
+MethodType replaceMethodType = MethodType.methodType(String.class, char.class, char.class);
+MethodHandle replaceMethodHandle = lookup.findVirtual(String.class, "replace", replaceMethodType);
+```
+
+In the next example, we are using `Lookup.findStatic` to look up a static method
+[`String.valueOf`](javadoc:String.valueOf(Object)), which accepts an `Object` and returns a `String`:
+
+```java
+MethodType valueOfMethodType = MethodType.methodType(String.class, Object.class);
+MethodHandle valueOfMethodHandle = lookup.findStatic(String.class, "valueOf", valueOfMethodType);
+```
+
+Similarly, we could use [`Lookup.findConstructor`](javadoc:MethodHandles.Lookup.findConstructor(Class,MethodType))
```

Comment:
I don't know whether it's necessary in the English language but better to unnecessarily note it than to overlook it.
```suggestion
Similarly, we could use the [`Lookup.findConstructor`](javadoc:MethodHandles.Lookup.findConstructor(Class,MethodType))
```